### PR TITLE
Removes unnecessary package installations in packaging pipelines

### PR DESCRIPTION
**Pull request by gurkanindibay**
With the recent changes in packaging images, linux package installations to execute validate_output is unnecessary now. 
In this PR, I removed them to make the pipeline more effective. 

- [x] Remove the test warning before merge

### CODE REVIEW

File: src/backend/distributed/commands/alter_table.c [hunk 1]
```diff
@@ -24,6 +24,8 @@
  *-------------------------------------------------------------------------
  */
 
+#warning "This is a compilation warning for testing purposes"
```
Comment (gurkanindibay) [inline src/backend/distributed/commands/alter_table.c:27]: Added this test warning to test validation phase with a valid warning

**Review by thanodnl — APPROVED** [review]
This is a great start.

`.github/packaging/validate_build_output.sh` still installs about 200MB (measured from my experimentation with folding everything into the containers) in pip packages.

Any chance we can add the cloning of citusdata/tools, the pyenv creation and the pip install to our packaging images to reduce the amount of external dependencies at runtime.

With 33 jobs fanning out we are downloading close to 7GB from pop on every-single-push. Lets donwload from container registries we control, and rely on caching possibilities in the infra, instead of external repositories.

Can be done in a separate PR. The fact that we don't need the other packages make me more confident in adding this to our images wihtout creating random failures, like I saw in almalinux 9

File: src/backend/distributed/commands/alter_table.c (hunk 1 above)
Comment (thanodnl) [reply src/backend/distributed/commands/alter_table.c:27]: Lets remove this and let tests pass

**Review by gurkanindibay — COMMENTED** [review]
Removes the test line

**Pull request merged**